parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -300,6 +402,52 @@ pub mod pallet {
 
 			Ok(true)
 		}
+
+		pub fn estimate_extra_fee(
+			ticket: &OutboundQueueTicket<MaxEnqueuedMessageSizeOf<T>>,
+		) -> Option<u128> {
+			match ticket.command.extra_fee_required() {
+				true => Some(
+					ticket
+						.command
+						.dispatch_gas()
+						.saturating_mul(GasPrice::<T>::get())
+						// Div by precision(decimal 10 for DOT and 18 for Ether)
+						.saturating_div(100_000_000)
+						.saturating_mul(SwapRatio::<T>::get()),
+				),
+				false => None,
+			}
+		}
+
+		pub fn estimate_base_fee(
+			ticket: &OutboundQueueTicket<MaxEnqueuedMessageSizeOf<T>>,
+		) -> Option<u128> {
+			match ticket.command.base_fee_required() {
+				true => Some(BaseFee::<T>::get()),
+				false => None,
+			}
+		}
+
+		pub fn charge_fees(
+			ticket: &OutboundQueueTicket<MaxEnqueuedMessageSizeOf<T>>,
+		) -> DispatchResult {
+			let agent_account = T::SovereignAccountOf::convert_location(&ticket.location)
+				.ok_or(Error::<T>::LocationToSovereignAccountConversionFailed)?;
+			// charge base fee here before https://github.com/paritytech/polkadot-sdk/pull/1234
+			// There is nuance difference between `xcm-fees-manager` because we charge from the
+			// agent account while they charge from the parachain account
+			let base_fee = Self::estimate_base_fee(ticket).unwrap_or_default();
+			let extra_fee = Self::estimate_extra_fee(ticket).unwrap_or_default();
+			let total_fee = base_fee.saturating_add(extra_fee).saturated_into::<BalanceOf<T>>();
+			T::Token::transfer(
+				&agent_account,
+				&T::LocalPalletId::get().into_account_truncating(),
```

Comment:
What happens to the fees collected in the Pallet account?

Reply:
We should actually be collecting the fees in BridgeHub's local treasury account, not the account for the pallet.

https://github.com/paritytech/polkadot-sdk/pull/1234/files#diff-984ede31f860995ed9e3720b323fce903524085f550b1a6c5a2a128664aa5846R64